logic/concludable/ConjunctionConcludable.java

```diff
@@ -112,21 +118,107 @@ public Value asValue() {
-            // Then build a Unification for each valid combination
-            return Stream.empty(); // TODO
+        public Stream<Map<Reference, Set<Reference>>> unify(ThenConcludable.Relation unifyWith) {
+            if (this.constraint().players().size() > unifyWith.constraint().players().size()) return Stream.empty();
+            Map<Reference, Set<Reference>> variableUnifier = new HashMap<>();
+            Optional<Map<Reference, Set<Reference>>> newUnifier;
+            if (constraint.owner().reference().isName()) {
+                newUnifier = tryExtendUnifier(this.constraint().owner(), unifyWith.constraint().owner(), variableUnifier);
+                if (newUnifier.isPresent()) variableUnifier = newUnifier.get();
+                else return Stream.empty();
+            }
+
+            if (constraint.owner().isa().isPresent() && constraint.owner().isa().get().type().reference().isName()) {
+                assert unifyWith.constraint().owner().isa().isPresent();
+                newUnifier = tryExtendUnifier(this.constraint().owner().isa().get().type(),
+                        unifyWith.constraint().owner().isa().get().type(), variableUnifier);
+                if (newUnifier.isPresent()) variableUnifier = newUnifier.get();
+                else return Stream.empty();
+            }
+
+            Map<Reference, Set<Reference>> finalVariableUnifier = variableUnifier;
+            List<RolePlayer> conjRolePlayers = Collections.unmodifiableList(constraint.players());
+            List<RolePlayer> thenRolePlayers = Collections.unmodifiableList(unifyWith.constraint().players());
+            return unifyRP(conjRolePlayers, thenRolePlayers, new HashMap<>(), 0)
+                    .map(unifier -> convertRolePlayerUnifier(unifier, thenRolePlayers, finalVariableUnifier));
+        }
+
+        private Stream<Map<RolePlayer, Set<Integer>>> unifyRP(
+                List<RolePlayer> conjRolePlayers, List<RolePlayer> thenRolePlayers,
+                Map<RolePlayer, Set<Integer>> unifier, Integer startCon) {
+
+            if (startCon == conjRolePlayers.size()) return Stream.of(unifier);
+            RolePlayer conjRP = conjRolePlayers.get(startCon);
+
+            return IntStream.range(0, thenRolePlayers.size())
+                    .filter(thenIdx -> unifier.values().stream().noneMatch(players ->
+                            players.contains(thenIdx)))
+                    .mapToObj(thenIdx ->
+                            tryExtendRolePlayerUnifier(conjRP, thenRolePlayers.get(thenIdx), thenIdx, unifier))
+                    .filter(Optional::isPresent).map(Optional::get)
+                    .flatMap(newUnifier ->
+                            unifyRP(conjRolePlayers, thenRolePlayers, newUnifier, startCon + 1));
+        }
+
+        private Map<Reference, Set<Reference>> convertRolePlayerUnifier(
+                Map<RolePlayer, Set<Integer>> rolePlayerUnifier, List<RolePlayer> thenRolePlayers,
```

Comment:
the first var should be called "unifiedRolePlayerIndices" or something right?

Reply:
Renamed to `mathcedRolePlayerIndices` in accord with the renaming outlined above.